courscala/src/main/scala/org/coursera/common/stringkey/StringKeyFormat.scala

```diff
@@ -165,6 +165,14 @@ object StringKeyFormat extends CommonStringKeyFormats {
       _ => StringKey(typeName))
   }
 
+  def typedCaseFormat[T, U](
```

Comment:
I'm a bit unsure about calling this "typed" since that nomenclature is not well-explained here. Could you either 1) rename to `prefixCaseFormat`, or 2) explain how this can be used for a type hierarchy of keys? Or both!